Linux support

lgleim opened the pull request:
This PR implements support for the Linux Platform and has been tested against the ~https://github.com/o3de/o3de/tree/stabilization/2210~ https://github.com/o3de/o3de/tree/stabilization/2205 (current main) branch

Commit:
Case Insensitive .gitignore

Commit:
Fix Vector type incompatability with Cesium Native  & Update deprecated CMake Macros

Commit:
Fix Import of Interpolator

Commit:
Fix Build of TilesetComponent
- Created default Destructur for TilesetComponent

Commit:
Add missing imports for OriginShiftAnchorComponentsBus

Commit:
Delete prohibited unused constants

Commit:
Fix QT Resource Initialization
- Worked around QT macro namespace limitations

Commit:
Added missing typename keywords

Commit:
Fix Linking issues due to relocated symbols
Enforce position independent code generation for Cesium Native using -fPIC compiler flag

Commit:
Augment Cmake scripts to find Cesium Native Libraries under Linux

Commit:
Explicit conversions between dvec3 <-> dvec4, dvec3 <-> fvec3, etc

Commit:
Applied Clang-format

lgleim commented:
Looking forward to your reviews @baothientran @shehzan10 @argallegos @kring :) Please also feel free to reach out to me on the O3DE discord if there is anything to discuss

baothientran commented:
Thank you @lgleim for the pull request! I will review it as soon as possible. I think there maybe a CLA that needs to be signed. @shehzan10 do you know where is the CLA to be signed?

shehzan10 commented:
@lgleim Thanks for your pull request, we appreciate your efforts! Cesium's CLA is available at https://github.com/CesiumGS/cesium/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla.

lgleim commented:
Thanks @shehzan10, I am reaching out to Open Source/Legal stakeholders internally to get the CLA signed. Might take a bit of time.

shehzan10 commented:
@lgleim Sounds good. FWIW, its a standard Apache Foundation derived CLA. Let me know if I can help in any way.

lgleim committed:
Update CMakeLists.txt

Unbreak the build of the current state of https://github.com/CesiumGS/cesium-o3de-samples

Commit:
Added Apache-2.0 Copyright Headers

Added Apache-2.0 copyright headers to all modified source files to
comply with the license requirements and Huawei's Open Source policy.

Commit:
Merge local changes

lgleim commented:
@shehzan10 CLA signed

shehzan10 commented:
@lgleim I can confirm that the CLA was signed. We'll take a look at this PR as soon as possible.

Thanks once again for your contribution!

cesium-concierge commented:
Thanks again for your contribution @lgleim! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

cesium-concierge commented:
Thanks again for your contribution @lgleim! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

cesium-concierge commented:
Thanks again for your contribution @lgleim! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

cesium-concierge commented:
Thanks again for your contribution @lgleim! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

cesium-concierge commented:
Thanks again for your contribution @lgleim! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

cesium-concierge commented:
Thanks again for your contribution @lgleim! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

cesium-concierge commented:
Thanks again for your contribution @lgleim! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

cesium-concierge commented:
Thanks again for your contribution @lgleim! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

cesium-concierge commented:
Thanks again for your contribution @lgleim! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

cesium-concierge commented:
Thanks again for your contribution @lgleim! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

cesium-concierge commented:
Thanks again for your contribution @lgleim! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

cesium-concierge commented:
Thanks again for your contribution @lgleim! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._
